src/System.Slices/System/Buffers/OwnedMemory.cs

```diff
@@ -4,6 +4,12 @@
 
 namespace System.Buffers
 {
+    public enum Mode {RefCount, LocalRefCount, None};
+
+    public class OwnedMemorySettings {
+        public static Mode s_mode = Mode.RefCount;
```

Comment:
If you make DisposableReservation non-generic, you could just add this to DisposableReservation, so there is no need for a new type. I know this type is probably just temporary (for experiments), but we try to minimize number of types in our APIs.

Also, s_mode does not follow our naming guidelines. Name it 'Mode'